src/components/elements/ConnectButton.tsx

```diff
@@ -1,48 +1,103 @@
-import React from "react";
 import { Button } from "@/components/ui/button";
-import { usePipecatClientTransportState } from "@pipecat-ai/client-react";
+import type { ButtonSize, ButtonVariant } from "@/components/ui/buttonVariants";
 import { cn } from "@/lib/utils";
+import { type TransportState } from "@pipecat-ai/client-js";
+import { usePipecatClientTransportState } from "@pipecat-ai/client-react";
+import React, { memo } from "react";
 
+/**
+ * Configuration for customizing button appearance and content for specific transport states.
+ * Allows partial configuration - only specify the states you want to customize.
+ */
+export type ConnectButtonStateContent = Partial<
+  Record<
+    TransportState,
+    {
+      /** The text or content to display in the button */
+      children: React.ReactNode;
+      /** The button variant to use for this state */
+      variant: ButtonVariant;
+      /** Optional CSS class to apply to the button */
+      className?: string;
+    }
+  >
+>;
+
+/**
+ * Props for the ConnectButton component.
+ */
 export type ConnectButtonProps = {
+  /** CSS class name to apply to the button */
   className?: string;
+  /** Callback function called when the connect action is triggered */
   onConnect?: () => void;
+  /** Generic click handler for the button */
   onClick?: () => void;
+  /** Callback function called when the disconnect action is triggered */
   onDisconnect?: () => void;
+  /** Size of the button component */
+  size?: ButtonSize;
+  /** Custom state content configuration for different transport states */
+  stateContent?: ConnectButtonStateContent;
 };
 
-export const ConnectButton: React.FC<ConnectButtonProps> = ({
+/**
+ * Internal component that renders a connect/disconnect button based on transport state.
+ * Handles button appearance, content, and interactions based on the current transport state.
+ *
+ * @param props - Component props including transport state and configuration
+ * @returns A button component that adapts to the current transport state
+ */
+export const ConnectButtonComponent: React.FC<
+  ConnectButtonProps & {
+    transportState: TransportState;
+  }
+> = ({
   className: passedClassName,
   onClick,
   onConnect,
   onDisconnect,
+  stateContent,
+  size = "default",
+  transportState,
 }) => {
-  const transportState = usePipecatClientTransportState();
-
+  /**
+   * Determines button properties based on current transport state and custom state content.
+   * Prioritizes custom state content over default behavior.
+   *
+   * @returns Button props including children, variant, and className
+   */
   const getButtonProps = (): React.ComponentProps<typeof Button> => {
+    // Check if we have custom content for this state
+    if (stateContent && stateContent[transportState]) {
+      return stateContent[transportState]!;
+    }
+
+    // Default content based on transport state
     switch (transportState) {
```

Comment:
Something here must have changed the ConnectButton behavior: after calling `initDevices()` the button switches to `Connecting…` and is disabled. You can see this in the 01-console example. It's not possible to connect to the agent anymore in this state.